### PR TITLE
Run tests in parallel with pytest-xdist

### PR DESCRIPTION
**Pull request by DRMacIver**
This used not to work, so this is somewhat experimental to see if it now does (and I'll fix it if it doesn't). Hopefully this should speed up a bunch of the build steps.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -119,13 +125,15 @@ commands=
...
     pytest==3.0.2
+    pytest-xdist==1.15.0
 commands=
     python -m pytest tests/pytest tests/cover/test_testdecorators.py
 
 [testenv:pytest28]
 basepython=python3.5
 deps =
     pytest==2.8.2
+    pytest-xdist==1.15.0
```
Comment (alexwlchan) [inline tox.ini]: Is there any way we can cut down repetition of dependency definitions in this file? 🤔

Comment (DRMacIver) [inline tox.ini]: Hmm. We could probably have a requirements-test.txt and use that. It causes tox some problems (it doesn't notice when the file has changed) but for our use case that's not a big deal given that we trash the tox environment before run.

File: appveyor.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ install:
   - "python --version"
   - "python -c \"import struct; print(struct.calcsize('P') * 8)\""
   - "%CMD_IN_ENV% python -m pip.__main__ install --upgrade setuptools pip"
-  - "%CMD_IN_ENV% python -m pip.__main__ install setuptools pytest==2.8.0 flaky"
+  - "%CMD_IN_ENV% python -m pip.__main__ install setuptools -rrequirements-test.txt
```
Comment (alexwlchan) [inline appveyor.yml]: You’re missing the closing quote here, which is causing Appveyor to fall over.

**Comment by DRMacIver** [comment]
BTW I've just noticed that this never got merged. @alexwlchan can you take a look at some point? It does help noticeably with build times.

File: requirements-test.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pytest==3.0.2
+pytest-xdist==1.15.0
+flaky
```
Comment (alexwlchan) [inline requirements-test.txt]: Given our issues with dependencies shifting silently in the past, any reason not to pin the version of flaky?

File: src/hypothesis/internal/debug.py [hunk 4]
```diff
@@ -87,12 +36,9 @@ def minimal(
...
-            settings=settings,
-            random=random,
-        )
-    return run()
+    return find(
+        definition,
+        condition,
+        settings=settings,
+        random=random,
+    )
```
Comment (alexwlchan) [inline src/hypothesis/internal/debug.py]: Look at all that lovely red. 😍

File: tox.ini [hunk 5]
```diff
@@ -29,17 +28,14 @@ commands =
...
+    -rrequirements-test.txt
 commands=
     python -m pytest tests/quality/
 
 [testenv:oldpy27]
 basepython=python2.7.3
 deps=
-  pytest==3.0.2
-  flaky
-  uncompyle6
```
Comment (alexwlchan) [inline tox.ini]: We’ve lost the uncompyle6 dependency – because it’s no longer used anywhere, yes?

File: tox.ini [hunk 6]
```diff
@@ -119,13 +115,15 @@ commands=
 basepython=python3.5
 deps =
     pytest==3.0.2
+    pytest-xdist==1.15.0
```
Comment (alexwlchan) [inline tox.ini]: This could call out to `-rrequirements-test.txt`.

File: tox.ini (hunk 5 above)
Comment (DRMacIver) [inline tox.ini]: Yeah. That was leftover cruft that was n o longer needed.

**Comment by DRMacIver** [comment]
Whoops. Sorry for delay in replying to changes. Should now all be done (but of course this is now blocked on the django issue)

**Pull request merged**